processor/memorylimiter/memorylimiter.go

```diff
@@ -149,12 +163,24 @@ func (ml *memoryLimiter) ConsumeMetricsData(
 	if ml.forcingDrop() {
 		numMetrics := len(md.Metrics)
 		stats.RecordWithTags(
-			context.Background(),
+			ctx,
 			ml.statsTags,
-			StatDroppedMetricCount.M(int64(numMetrics)))
+			processor.StatDroppedMetricCount.M(int64(numMetrics)),
+			processor.StatMetricBatchesDroppedCount.M(1))
+
+		// TODO: actually to be 100% sure that this is "refused" and not "dropped"
+		// 	it is necessary to check the pipeline to see if this is directly connected
+		// 	to a receiver (ie.: a receiver is on the call stack). For now it
+		// 	assumes that the pipeline is properly configured and a receiver is on the
+		// 	callstack.
```

Comment:
Again I am a bit confused what exactly "refused" means.

Also, I would imagine "dropped" is only counted when we return success to our predecessor but decide not to forward the data to the next component. If that is not how "dropped" is defined then please clarify.

Reply:
Your understanding of "dropped" is correct. Let me see if I can rewrite to make it more clear.